src/olympia/files/models.py

```diff
@@ -76,6 +77,13 @@ def files_storage():
 class File(OnChangeMixin, ModelBase):
     id = PositiveAutoField(primary_key=True)
     STATUS_CHOICES = amo.STATUS_CHOICES_FILE
+    STATUS_DISABLED_REASONS = Choices(
+        ('NONE', 0, 'None'),
+        ('DEVELOPER', 1, 'Developer disabled'),
+        ('ADDON_DISABLE', 2, 'Add-on disabled'),
+        ('ADDON_DELETE', 3, 'Add-on deleted'),
+        ('VERSION_DELETE', 4, 'Version deleted'),
```

Comment:
we don't really need to know about deletes for this patch, but it seemed consistent to record all the ways an approved or awaiting review file could change to disabled.  
(At one point I was recording _all_ the reasons a file would be STATUS_DISABLED - rejection and obsolete too - but it made the patch overly complicated and didn't add any functionality we actually need)